### DISCUSSION TITLE
Keycloak dont run docker Compose

### DISCUSSION THREAD

**Discussion by UsefullWolf** [keycloak/keycloak#21311] [Q&A] [answered]
I am having problems running keycloak on docker.

`version: '3.8'
services:
  keycloak:
    image: quay.io/keycloak/keycloak:latest
    ports:
      - 9210:8080
    environment:
      - DB_VENDOR=postgres
      - DB_ADDR=postgres
      - DB_DATABASE=keycloak
      - DB_USER=keycloak
      - DB_PASSWORD=keycloak
      - KC_HTTPS_CERTIFICATE_FILE=/root/docker/keycloak/ssl/cert.pem
      - KC_HTTPS_CERTIFICATE_KEY_FILE=/root/docker/keycloak/ssl/key.pem
    depends_on:
      - postgres
    command:
      - start --auto-build --hostname-url=https://auth.wolfsrealm.org --optimized --https-certificate-file=/root/docker/keycloak/ssl/cert.pem --https-certificate-key-file=/root/docker/keycloak/ssl/key.pem
  postgres:
    image: postgres:latest
    environment:
      - POSTGRES_DB=keycloak
      - POSTGRES_USER=keycloak
      - POSTGRES_PASSWORD=keycloak
    volumes:
      - ./postgres/data:/var/lib/postgresql/data
      - /root/docker/keycloak/ssl:/ssl:ro
`

`65-7589: no members discovered after 2001 ms: creating cluster as coordinator
keycloak-keycloak-1  | 2023-06-29 09:25:40,502 INFO  [org.infinispan.CLUSTER] (keycloak-cache-init) ISPN000094: Received new cluster view for channel ISPN: [39fc57f60b65-7589|0] (1) [39fc57f60b65-7589]
keycloak-keycloak-1  | 2023-06-29 09:25:40,505 INFO  [org.infinispan.CLUSTER] (keycloak-cache-init) ISPN000079: Channel `ISPN` local address is `39fc57f60b65-7589`, physical addresses are `[172.20.0.3:40551]`
keycloak-keycloak-1  | 2023-06-29 09:25:40,800 INFO  [org.keycloak.connections.infinispan.DefaultInfinispanConnectionProviderFactory] (main) Node name: 39fc57f60b65-7589, Site name: null
keycloak-keycloak-1  | 2023-06-29 09:25:40,911 INFO  [org.keycloak.broker.provider.AbstractIdentityProviderMapper] (main) Registering class org.keycloak.broker.provider.mappersync.ConfigSyncEventListener
keycloak-keycloak-1  | 2023-06-29 09:25:41,323 INFO  [org.infinispan.CLUSTER] (main) ISPN000080: Disconnecting JGroups channel `ISPN`
keycloak-keycloak-1  | 2023-06-29 09:25:41,380 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Failed to start server in (production) mode
keycloak-keycloak-1  | 2023-06-29 09:25:41,381 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: /root/docker/keycloak/ssl/cert.pem
keycloak-keycloak-1  | 2023-06-29 09:25:41,382 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) Key material not provided to setup HTTPS. Please configure your keys/certificates or start the server in development mode.
keycloak-keycloak-1  | 2023-06-29 09:25:41,382 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) For more details run the same command passing the '--verbose' option. Also you can use '--help' to see the details about the usage of the particular command.
keycloak-keycloak-1 exited with code 1
`

**Comment by UsefullWolf** [comment]
version: '3.8'
services:
  keycloak:
    image: quay.io/keycloak/keycloak:latest
    ports:
      - 9210:8080
    environment:
      - DB_VENDOR=postgres
      - DB_ADDR=postgres
      - DB_DATABASE=keycloak
      - DB_USER=keycloak
      - DB_PASSWORD=keycloak
      - KC_HTTPS_CERTIFICATE_FILE=/root/docker/keycloak/ssl/cert.pem
      - KC_HTTPS_CERTIFICATE_KEY_FILE=/root/docker/keycloak/ssl/key.pem
    volumes:
      - /root/docker/keycloak/ssl:/ssl:ro
    depends_on:
      - postgres
    command:
      - start --auto-build --hostname-url=https://auth.wolfsrealm.org --optimized --https-certificate-file=/root/docker/keycloak/ssl/cert.pem --https-certificate-key-file=/root/docker/keycloak/ssl/key.pem
  postgres:
    image: postgres:latest
    environment:
      - POSTGRES_DB=keycloak
      - POSTGRES_USER=keycloak
      - POSTGRES_PASSWORD=keycloak
    volumes:
      - ./postgres/data:/var/lib/postgresql/data

**Reply by adnanvrbic** [reply]
Please check this mapping (- /root/docker/keycloak/ssl:/ssl:ro), this doesn't look correct. If you ssh to container can you see the pem files?

**Reply by UsefullWolf** [reply]
what about it doesnt look correct?

**Reply by adnanvrbic** [reply, marked as answer]
Actually, I see that :ro part will be ignored, but if you are mapping local folder /root/docker/keycloak/ssl to some other folder in docker image, than this needs to be changed to the docker image path:

- KC_HTTPS_CERTIFICATE_FILE=/root/docker/keycloak/ssl/cert.pem
- KC_HTTPS_CERTIFICATE_KEY_FILE=/root/docker/keycloak/ssl/key.pem

in example: 

volumes:
- /root/docker/keycloak/ssl:/tmp/data

then configuration should be something like this:
- KC_HTTPS_CERTIFICATE_FILE=/tmp/data/cert.pem
- KC_HTTPS_CERTIFICATE_KEY_FILE=/tmp/data/key.pem

I'm posting all this as error that you provided means that keycloak can't find pem files.

**Reply by UsefullWolf** [reply]
ok ill try

**Reply by UsefullWolf** [reply]
<3 thanks

**Comment by UsefullWolf** [comment]
version: '3.8'
services:
  keycloak:
    image: quay.io/keycloak/keycloak:latest
    ports:
      - 9210:8443
    environment:
      - DB_VENDOR=postgres
      - DB_ADDR=postgres
      - DB_DATABASE=keycloak
      - DB_USER=keycloak
      - DB_PASSWORD=keycloak
      - KC_HTTPS_CERTIFICATE_FILE=/tmp/data/cert.pem
      - KC_HTTPS_CERTIFICATE_KEY_FILE=/tmp/data/key.pem
      - KC_HOSTNAME=auth.wolfsrealm.org
    volumes:
      - /root/docker/keycloak/ssl:/tmp/data
    depends_on:
      - postgres
    command:
      - start --optimized
#--hostname-url=https://auth.wolfsrealm.org --optimized --https-certificate-file=/root/docker/keycloak/ssl/cert.pem --https-certificate-key-file=/root/docker/keycloak/ssl/key.pem
  postgres:
    image: postgres:latest
    environment:
      - POSTGRES_DB=keycloak
      - POSTGRES_USER=keycloak
      - POSTGRES_PASSWORD=keycloak
    volumes:
      - ./postgres/data:/var/lib/postgresql/data

here a now running config
